bokeh/io.py

```diff
@@ -310,7 +315,12 @@ def _show_with_state(obj, state, browser, new, notebook_handle=False):
     shown = False
 
     if state.notebook:
-        comms_handle = _show_notebook_with_state(obj, state, notebook_handle)
+        if state.notebook_name.lower() == "jupyter":
+            comms_handle = _show_notebook_with_state(obj, state, notebook_handle)
+        elif state.notebook_name.lower() == "zeppelin":
+            _show_zeppelin(obj)
+        else:
+            raise Exception("Not supported notebook type:" + state.notebook_name)
```

Comment:
This should be handled by having a validating property setter, as described above. Then `state.notebook_type` is known to always be good (and no `.lower` is needed either)